### PR TITLE
Update Fake.Tools.Pickles

### PR DESCRIPTION
**Pull request by magicmonty**
### Description

Pickles has gotten some additional command line parameters 
I added them and updated the command line generation from a StringBuilder to 
`Args.toWindowsCommandLine`

## TODO

- [x] New (API-)documentation for new features exist
- [x] tested manually
- [x] Fake 5 [API guideline](https://fake.build/contributing.html#API-Design) is honored

### CODE REVIEW

**Comment by magicmonty** [comment]
@matthid Added unit tests and fixed the resulting bugs ;)

**Comment by matthid** [comment]
Tried to fix on the phone but I failed :(

File: src/test/Fake.Core.UnitTests/Fake.Tools.Pickles.fs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  let file, args =
+    match cp.Command with
+    | RawCommand(file, args) -> file, args
+    | _ -> failwithf "expected RawCommand"
+    |> ArgumentHelper.checkIfMono
+    
+  let expectedPath = Path.Combine("pickles", "pickles.exe")
+  Expect.equal file expectedPath "Expected pickles.exe"
+  
+  expectedPath, cp.Command.CommandLine
```
Comment (matthid) [inline src/test/Fake.Core.UnitTests/Fake.Tools.Pickles.fs]: this needs to use the 'args' variable and not 'cp' in order to work with mono environments...

I think '(RawCommand(file, args)).CommandLine' should work

Can you please change the other ones as well, this should make the build green again

**Comment by magicmonty** [comment]
@matthid It seems now to work except for that the check for mono fails, if there are no arguments at all

**Comment by matthid** [comment]
Ah yes, thanks for taking care of this!

**Comment by matthid** [comment]
Yes thanks a lot :)

**Pull request merged**